sir_cake/seller/views.py

```diff
@@ -79,3 +81,40 @@ def delete_item(request, pk):
         messages.success(request, "Item successfully deleted")
         return redirect('all_items')
     return render(request, 'seller/delete-item.html', {'item': item})
+
+
+def orders(request):
+    orders = Order.objects.exclude(
+        status='preorder').order_by('-order_date')
+
+    order_status = request.GET.get('status')
+    if order_status is None:
+        order_status = 'ALL'
+    if order_status == 'UNSENT':
+        orders = orders.filter(status='paid')
+    if order_status == 'SHIPPED':
+        orders = orders.filter(status='shipped')
+
+    orders_and_prices = []
+    for order in orders:
+        sum_price = total_order_price(BasketItem.objects.filter(order=order))
+        orders_and_prices.append((order, sum_price))
+
+    if request.method == 'POST':
```

Comment:
If you are handling a POST request, you are again executing some code before this `if` statement that doesn't need to be executed :). This could probably be refactored a little bit